### ISSUE TITLE
Unable to change routes with latest version

### ISSUE THREAD

**Issue by RikuVan** [hyperapp/router#16]
I updated to
```
"@hyperapp/router": "^0.2.3",
 "hyperapp": "^0.12.0",
```
Initial page renders but I get the following error when changing routes: 
```
Uncaught TypeError: Cannot read property '1' of undefined
    at render (router.js:1)
    at hyperapp.js:1
    at Array.map (<anonymous>)
    at i (hyperapp.js:1)
    at t (hyperapp.js:1)
```
Here is the view
```
view: [
    ['/', (state, actions) =>
      <Main
        state={state}
        add={actions.add}
        setValue={actions.setValue}
        router={actions.router}
      />
    ],
    [
      '/previous-events',
      (state, actions) =>
        <PreviousEvents
          html={state.html}
          loading={state.showSpinner}
          a={actions}
        />
    ],
    ['*', (state, actions) => actions.router.go('/')],
  ]
```
[The code with old deps can be found](https://github.com/webDevAndSausages/webDevAndSausages.github.io)

**Comment by jbucaran** [comment]
@RikuVan What is this? 

```js
 ['*', (state, actions) => actions.router.go('/')]
```

**Comment by RikuVan** [comment]
Yeah, stupid should just be the default view. But I don't think that is the bug.

**Comment by jbucaran** [comment]
@RikuVan Can you post a full example? Otherwise I can't debug this.

**Comment by RikuVan** [comment]
[code is here](https://github.com/webDevAndSausages/webDevAndSausages.github.io). There is not much too it really

**Comment by jbucaran** [comment]
I can't reproduce your bug.

```js
app({
  view: [
    ["/previous-events", (state, actions) => <h1>previous-events</h1>],
    ["*", (state, actions) => <h1>wildcard</h1>]
  ],
  mixins: [Router]
})
```

<img width="400" alt="screen shot 2017-08-24 at 2 34 22" src="https://user-images.githubusercontent.com/56996/29629697-c342d25a-8874-11e7-9172-2db37bf34acb.png">
<img width="400" alt="screen shot 2017-08-24 at 2 34 20" src="https://user-images.githubusercontent.com/56996/29629698-c36dfc82-8874-11e7-9b3d-83f746b9ae44.png">

**Comment by jbucaran** [comment]
@RikuVan I couldn't repro this. Can you post an example I can copy & paste?

**Comment by RikuVan** [comment]
As soon as i have a chance I will debug a bit more myself first. Are you downloading router from npm or using master?

**Comment by jbucaran** [comment]
I am using master.

**Comment by RikuVan** [comment]
so it seems to work if I take the hyphen out of the route name: "previous_events" works but "previous-events" does not. So maybe the regex needs some tweaking?

**Comment by jbucaran** [comment]
@RikuVan `previous-events` worked in my local test.

**Comment by jbucaran** [comment]
https://github.com/hyperapp/router/blob/13fbefe169d820c7b326ae17a7c9e73af01ad187/src/Router.js#L56

**Comment by RikuVan** [comment]
oh I was pretty sure you would get the same error with hyphen :anguished: Dunno. Will look again tomorrow if I can.

**Comment by jbucaran** [comment]
@SkaterDad Can you repro this?

**Comment by SkaterDad** [comment]
I'll give it a try later tonight or tomorrow as I get time. 

What browser is the bug occurring in?

**Comment by RikuVan** [comment]
both chrome and firefox

**Comment by jbucaran** [comment]
@RikuVan Can you create a different, small example where you can repro the bug?

**Comment by RikuVan** [comment]
I think I figured out what the problem was. I logged out the router data in the setter and it was a promise when hitting the previous-events route. So I suppose just like the load event you should not return anything from the route action. It was nothing to do with the hyphen. Interesting that this caused no bug with older version of router and hyperapp. Should of debugged this further before making an issue. Thanks for your help.

**Comment by jbucaran** [comment]
> @RikuVan  So I suppose just like the load event you should not return anything from the route action.

The route events hooks into the route switch process, essentially letting you override the route that is going to match for another one. But is that really useful?

We could just get rid of that no-feature or better document it.

Here is the relevant part.

https://github.com/hyperapp/router/blob/13fbefe169d820c7b326ae17a7c9e73af01ad187/src/Router.js#L29-36

**Comment by RikuVan** [comment]
Seems it could be useful. But, yeah, maybe a warning in the docs about how to use/not use it properly.

**Comment by RikuVan** [comment]
Uses cases probably include something like a. a first time user redirect to a orientation page or auth stuff or maybe changing the locale. But I guess a HOF would be my choice for auth stuff.

**Comment by RikuVan** [comment]
oops just to confuse things more it was the route 'event' that i returned from not the action. So to confirm: you shouldn't return anything from the route event correct?

**Comment by jbucaran** [comment]
> @RikuVan It was the route 'event' that i returned from not the action.

Yeah, that's what I thought and so is my answer.

> So to confirm: you shouldn't return anything from the route event correct?

You can return a [RouteInfo](https://github.com/hyperapp/router#routeinfo) object to use instead, but we're still yet to see if that's even useful.

BTW what were you returning from events.route and why? Just curious.

**Comment by RikuVan** [comment]
I just unnecessarily added a return statement to a axios promise out of habit. And yeah i had heard about hyperapp for the first time the same day I coded, so I was guessing about everything. In a way, it was nice to have this bug because it got me look at your code. I should do the same for hyperapp now (maybe with the next bug :). So cool when the 'big' frameworks are a blackbox to me to be able to stick a debugger statement in the source and follow. Thanks to hyperapp.

**Comment by jbucaran** [comment]
Cool. I'll keep this open until I get around adding more docs about `route`.

**Comment by JorgeBucaran** [comment]
@RikuVan I just published 0.4.1; it works with hyperapp@0.18.0. Docs are on the README. 👋
